### PR TITLE
feat: Adds a new metadata bundle type for AppFrameworkTemplates

### PR DESCRIPTION
**Pull request by rvwatch**
### What does this PR do?
Adds a new metadata bundle type for AppFrameworkTemplates. 

### What issues does this PR fix or reference?
@W-16928554@

#<Insert GitHub Issue>, @<Insert GUS WI>@

### Functionality Before
CLI no have AppFrameworkTemplates metadata
<insert gif and/or summary>

### Functionality After
CLI now have AppFrameworkTemplates metadata!
<insert gif and/or summary>

### CODE REVIEW

File: src/registry/metadataRegistry.json [hunk 1]
```diff
@@ -4639,6 +4644,49 @@
       "directoryName": "extlClntAppPushPolicies",
       "inFolder": false,
       "strictDirectoryName": false
+    },
+    "appframeworktemplatebundle": {
```
Comment (rvwatch) [inline src/registry/metadataRegistry.json:4648]: I only added this guy. Scripts added the rest.

File: src/utils/filePathGenerator.ts [hunk 2]
```diff
@@ -112,17 +112,26 @@ export const filePathsFromMetadataComponent = (
 
   // lwc, aura, waveTemplate, experiencePropertyType
   if (type.strategies?.adapter === 'bundle') {
-    const mappings = new Map<string, string>([
+    const mappings = new Map<string, string | string[]>([
```
Comment (rvwatch) [inline src/utils/filePathGenerator.ts:115]: I didn't want to mess around with the other bundles, so the only way I could get this to accept my new bundle and still pass the tests was to add the or string[].

File: src/utils/filePathGenerator.ts [hunk 3]
```diff
@@ -112,17 +112,26 @@ export const filePathsFromMetadataComponent = (
...
+      ],
     ]);
+
     const matched = mappings.get(type.name);
     if (!matched) {
       throw messages.createError('unsupportedBundleType', [type.name]);
     }
-    return [matched];
+
+    return Array.isArray(matched) ? matched : [matched];
```
Comment (rvwatch) [inline src/utils/filePathGenerator.ts:134]: Dealing with my array of strings or strings situation.

File: src/registry/metadataRegistry.json (hunk 1 above)
Comment (reidaelliott) [reply src/registry/metadataRegistry.json:4648]: might want to check with the cli team on these additions.  Is it normal for scripts to add unrelated stuff. Leave or remove?

**Review by WillieRuemmele — APPROVED** [review]

**Pull request merged**